### PR TITLE
Extract CatalogClient from flytepropeller to flytestdlib

### PR DESCRIPTION
**Pull request by MorpheusXAUT**
# TL;DR
The `CatalogClient` implementation has been extracted from flytepropeller's source code and moved to flytestdlib as flyteadmin also needs access to the functionality.

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
The whole latest implementation of the catalog package was moved from flytepropeller to [flytestdlib](https://github.com/flyteorg/flytestdlib/pull/148) since flyteadmin will need access to a client interacting with datacatalog for evicting cached executions.

The PR was created as a draft until the [flyteidl](https://github.com/flyteorg/flyteidl/pull/353), [flyteplugins](https://github.com/flyteorg/flyteplugins/pull/301) and [flytestdlib](https://github.com/flyteorg/flytestdlib/pull/148) PRs have been merged and respective versions have been published.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2867

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by MorpheusXAUT** [comment]
Rebased onto current `master` branch.
